File: app-java-fx/src/main/java/chessgame/app/fx/controller/BoardController.java
package chessgame.app.fx.controller;

import chessgame.app.fx.view.PieceView;
import chessgame.facade.BoardFacade;
import chessgame.facade.PieceFacade;
import chessgame.facade.mock.MockBoardFacade;
import javafx.fxml.FXML;
import javafx.scene.layout.GridPane;
import javafx.scene.layout.Pane;

import java.util.Map;

/**
 * Controller of Chessboard.
 *
 * @author Prestes, E. M.
 * @see GridPane
 * @see AbstractController
 * @see FXML
 * @since October 2016
 */
public class BoardController extends AbstractController {

    public static final String VIEW_TITLE = "Chess Game";

    public static final String VIEW_URL = String.format(VIEW_PACKAGE, "BoardView");

    private final BoardFacade facade = new MockBoardFacade();

    @FXML
    public GridPane boardView;

    @FXML
    public void initialize() {
        initialize(facade.initBlackPieces());
        initialize(facade.initWhitePieces());
    }

    private void initialize(Map<String, PieceFacade> pieces) {
        pieces.forEach((position, piece) -> boardView.getChildren().stream()
                .filter(blockView -> position.equals(blockView.getId()))
                .map(foundView -> ((Pane) foundView).getChildren())
                .forEach(foundView -> foundView.add(PieceView.create(piece))));
    }
}

File: app-kotlin-swing/src/main/kotlin/chessgame/app/swing/controller/BoardController.kt
package chessgame.app.swing.controller

import chessgame.app.swing.Controller
import chessgame.app.swing.view.BoardView
import chessgame.facade.BoardFacade
import chessgame.facade.mock.MockBoardFacade
import java.awt.Dimension

class BoardController(
        override val view: BoardView = BoardView(COLS_LABEL, ROWS_LABEL),
        override val size: Dimension= Dimension(500, 500),
        override val title: String = "Chess Game",
        override val labelWest: Array<String> = ROWS_LABEL,
        override val labelEast: Array<String> = ROWS_LABEL,
        override val labelNorth: Array<String> = COLS_LABEL,
        override val labelSouth: Array<String> = COLS_LABEL
) : Controller {

    private val facade: BoardFacade = MockBoardFacade()

    override fun initMatch() {
        with(facade) {
            initPieces(initBlackPieces())
            initPieces(initWhitePieces())
        }
    }

    private fun initPieces(pieces: Map<String, String>) {
        pieces.forEach { (pos, piece) -> view.add(pos, piece) }
    }

    companion object {
        private val COLS_LABEL = arrayOf("A", "B", "C", "D", "E", "F", "G", "H")
        private val ROWS_LABEL = arrayOf("1", "2", "3", "4", "5", "6", "7", "8")
    }
}
